jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -64,14 +66,18 @@ YUI.add('user-profile', function() {
       });
     },
 
-    componentDidUpdate: function(prevProps, prevState) {
+    componentWillReceiveProps: function(nextProps) {
```

Comment:
I think that this needs a test, especially because it caused us such a large issue. Let me know if you need some help putting it together.